### ISSUE TITLE
Add Command cannot find script?

### ISSUE THREAD

**Issue by Alamantus** [nextcloud/spreed#4575]
## Steps to reproduce
1. Install NextCloud with Snap and get it working correctly
1. Create script at `/usr/local/bin/nextcloud-command-roll.sh` with correct read and execute access
1. Try to add it as a command using `sudo nextcloud.occ talk:command:add roll Rolly "/usr/local/bin/nextcloud-command-roll.sh {ARGUMENTS} {ROOM} {USER}" 2 2`

### Expected behaviour
It should add the script.

### Actual behaviour
It returns `sh: 1: /usr/local/bin/nextcloud-command-roll.sh: not found` and `The script is invalid`.

Running `ls -lah /usr/local/bin` returns

```
total 32K
drwr-xr-x  2 root root 4.0K Nov 10 18:35 .
drwr-xr-x 10 root root 4.0K Apr 23  2020 ..
-rwxr-xr-x  1 root root 1.1K Nov 10 18:35 nextcloud-command-roll.sh
...........
```

which shows that the script does in fact exist...

I also tried moving the script to the data folder, and it says `Permission denied` instead of `not found`, which feels like progress, but I don't know how to give it the permission it needs.

## Talk app

**Talk app version:** 10.0.1

**Custom Signaling server configured:** no

**Custom TURN server configured:** yes

**Custom STUN server configured:** yes

## Browser

N/A

### Browser log

N/A

## Server configuration

**Operating system**: Ubuntu

**Web server:** Apache? (Installed with Snap)

**Database:** Maria? (Installed with Snap)

**PHP version:** 7.4.3

**Nextcloud Version:** 20.0.1

**List of activated apps:**

<details>
```
Enabled:
  - accessibility: 1.6.0
  - activity: 2.13.2
  - apporder: 0.11.0
  - bookmarks: 4.0.5
  - bruteforcesettings: 2.0.1
  - calendar: 2.1.2
  - cloud_federation_api: 1.3.0
  - comments: 1.10.0
  - contacts: 3.4.1
  - contactsinteraction: 1.1.0
  - dashboard: 7.0.0
  - dav: 1.16.0
  - federatedfilesharing: 1.10.1
  - federation: 1.10.1
  - files: 1.15.0
  - files_pdfviewer: 2.0.1
  - files_rightclick: 0.17.0
  - files_sharing: 1.12.0
  - files_trashbin: 1.10.1
  - files_versions: 1.13.0
  - files_videoplayer: 1.9.0
  - firstrunwizard: 2.9.0
  - guests: 1.6.1
  - impersonate: 1.7.0
  - logreader: 2.5.0
  - lookup_server_connector: 1.8.0
  - metadata: 0.12.0
  - music: 0.17.3
  - news: 15.0.6
  - nextcloud_announcements: 1.9.0
  - notifications: 2.8.0
  - oauth2: 1.8.0
  - password_policy: 1.10.1
  - photos: 1.2.0
  - privacy: 1.4.0
  - provisioning_api: 1.10.0
  - serverinfo: 1.10.0
  - settings: 1.2.0
  - sharebymail: 1.10.0
  - spreed: 10.0.1
  - support: 1.3.0
  - survey_client: 1.8.0
  - systemtags: 1.10.0
  - tasks: 0.13.5
  - text: 3.1.0
  - theming: 1.11.0
  - theming_customcss: 1.7.0
  - twofactor_backupcodes: 1.9.0
  - user_status: 1.0.0
  - viewer: 1.4.0
  - weather_status: 1.0.0
  - workflowengine: 2.2.0
Disabled:
  - admin_audit
  - encryption
  - end_to_end_encryption
  - files_external
  - recommendations
  - user_ldap
```
</details>

**Nextcloud configuration:**

<details>
```
{
    "system": {
        "apps_paths": [
            {
                "path": "\/snap\/nextcloud\/current\/htdocs\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/snap\/nextcloud\/current\/nextcloud\/extra-apps",
                "url": "\/extra-apps",
                "writable": true
            }
        ],
        "supportedDatabases": [
            "mysql"
        ],
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "log_type": "file",
        "logfile": "\/var\/snap\/nextcloud\/current\/logs\/nextcloud.log",
        "logfilemode": 416,
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "localhost"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "20.0.1.1",
        "overwrite.cli.url": "http:\/\/localhost",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "hub.alamantus.com",
        "overwriteprotocol": "https",
        "mail_smtpmode": "smtp",
        "mail_smtpsecure": "ssl",
        "mail_sendmailmode": "smtp",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "app_install_overwrite": [
            "keeporsweep"
        ],
        "loglevel": 0
    }
}
```
</details>

**Comment by nickvergessen** [comment]
Try chown www-data

**Comment by Alamantus** [comment]
> 
> 
> Try chown www-data

I still get `Permission denied`.

My `data` folder _is_ on an external hard drive that gets auto-mounted to `/media/nextcloud`, and I _have_ given Snap the permission it needs to use that folder. I don't know if that makes any difference for this.

**Comment by Alamantus** [comment]
I ended up moving the script to `/var/snap/nextcloud/common/nextcloud/scripts` (I created the `scripts` folder) and used `sudo chown www-data roll.sh` inside that folder and the command was added successfully.

Feel free to close this issue because I ended up getting what I wanted, but the original question isn't answered, re: the script not being able to find the file in the filesystem. That is probably outside of the scope of the Spreed project, though.
